zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -361,7 +361,42 @@ where
         }?
     }
 
-    let (tx, tx_metadata) = builder.build(&prover).map_err(Error::Builder)?;
+    let fee_strategy = BasicFixedFeeChangeStrategy::new(DEFAULT_FEE);
+    let balance = fee_strategy
+        .compute_balance(
+            params,
+            target_height,
+            builder.transparent_inputs(),
+            builder.transparent_outputs(),
+            builder.sapling_inputs(),
+            builder.sapling_outputs(),
+        )
+        .map_err(|e| match e {
+            ChangeError::InsufficientFunds {
+                available,
+                required,
+            } => Error::InsufficientBalance(available, required),
+            ChangeError::StrategyError(e) => Error::BalanceError(e),
+        })?;
```

Comment:
We do this match in two places, so we should add an `impl From<ChangeError> for Error`.

Reply:
Error handling changes *substantially* in #689 so let's wait to alter this.